### PR TITLE
Feature/116 Sidebar refactor

### LINKED ISSUE #116: Refactor sidebar
- [ ] Community
- [ ] Help us
- [ ] Statistics
- [ ] Simplify vndb subtitle

### PR DESCRIPTION
**Pull request by moetazaneta**
resolves #116 

How can I do `getLink` in sidebar better?

### CODE REVIEW

File: src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -11,6 +11,52 @@ interface Props {
   readonly onSidebarHide: () => void;
 }
 
+/**
+ * Gets link element.
+ * @param element Sidebar link element.
+ * @returns
```
Comment (NightOfNights) [inline src/components/Sidebar/Sidebar.tsx]: Remove `@returns`

File: src/components/Sidebar/data.ts [hunk 2]
```diff
@@ -7,17 +7,34 @@ export enum SidebarElementType {
...
+export type SidebarLinkElement = {
 
   /** Type. */
   readonly type: SidebarElementType.Link;
 
   /** Text. */
   readonly text: string;
 
+    /** Whether link is external. */
+    readonly isDisabled?: boolean;
```
Comment (NightOfNights) [inline src/components/Sidebar/data.ts]: Incorrect indentation

File: src/components/Sidebar/data.ts [hunk 3]
```diff
@@ -7,17 +7,34 @@ export enum SidebarElementType {
...
+
+} & (
+  {
+
+  /** Whether link is external. */
```
Comment (NightOfNights) [inline src/components/Sidebar/data.ts:9]: I understand what you are doing here, but can you improve code-readability, if possible? 🙂

**Review by NightOfNights — COMMENTED** [review]
🥲
image

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.suggest.jsdoc.generateReturns": false
+}
```
Comment (prekel) [inline .vscode/settings.json:3]: is it needed?

**Review by prekel — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 4 above)
Comment (moetazaneta) [reply .vscode/settings.json:3]: I don't think that should be add to repo, but I'll use it locally, ty

**Review by NightOfNights — APPROVED** [review]

**Review by Moltenship — APPROVED** [review]
These layout jumps seems strange to me

**Pull request merged**